src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -212,29 +190,112 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
+                    services.TryAddExisting<IGrainFactory, GrainFactory>();
+                    services.TryAddExisting<IInternalGrainFactory, GrainFactory>();
+                    services.TryAddSingleton<TypeMetadataCache>();
+                    services.TryAddSingleton<AssemblyProcessor>();
+                    services.TryAddSingleton<ActivationDirectory>();
+                    services.TryAddSingleton<LocalGrainDirectory>();
+                    services.TryAddExisting<ILocalGrainDirectory, LocalGrainDirectory>();
+                    services.TryAddSingleton<SiloStatisticsManager>();
+                    services.TryAddSingleton<ISiloPerformanceMetrics>(sp => sp.GetRequiredService<SiloStatisticsManager>().MetricsTable);
+                    services.TryAddSingleton<SiloAssemblyLoader>();
+                    services.TryAddSingleton<GrainTypeManager>();
+                    services.TryAddExisting<IMessagingConfiguration, GlobalConfiguration>();
+                    services.TryAddSingleton<MessageCenter>();
+                    services.TryAddExisting<IMessageCenter, MessageCenter>();
+                    services.TryAddExisting<ISiloMessageCenter, MessageCenter>();
+                    services.TryAddSingleton<Catalog>();
+                    services.TryAddSingleton(sp => sp.GetRequiredService<Catalog>().Dispatcher);
+                    services.TryAddSingleton<InsideRuntimeClient>();
+                    services.TryAddExisting<IRuntimeClient, InsideRuntimeClient>();
+                    services.TryAddSingleton<MembershipFactory>();
+                    services.TryAddSingleton<MultiClusterOracleFactory>();
+                    services.TryAddSingleton<LocalReminderServiceFactory>();
+                    services.TryAddSingleton<DeploymentLoadPublisher>();
+                    services.TryAddExisting<IDeploymentLoadPublisher, DeploymentLoadPublisher>();
+                    services.TryAddSingleton<IMembershipTable>(
+                        sp => sp.GetRequiredService<MembershipFactory>().GetMembershipTable(sp.GetRequiredService<GlobalConfiguration>()));
+                    services.TryAddSingleton<MembershipOracle>(
+                        sp =>
+                        sp.GetRequiredService<MembershipFactory>()
+                          .CreateMembershipOracle(sp.GetRequiredService<Silo>(), sp.GetRequiredService<IMembershipTable>()));
+                    services.TryAddExisting<IMembershipOracle, MembershipOracle>();
+                    services.TryAddExisting<ISiloStatusOracle, MembershipOracle>();
+                    services.TryAddSingleton<Func<ISiloStatusOracle>>(sp => () => sp.GetRequiredService<ISiloStatusOracle>());
+                    services.TryAddSingleton<MultiClusterOracleFactory>();
+                    services.TryAddSingleton<LocalReminderServiceFactory>();
+                    services.TryAddSingleton<SiloControl>();
```

Comment:
Very good point. I'll use `ActivatorUtilities` as you suggest.

Later on, I'll likely change this for IMembershipOracle so that if the user has injected their own implementation (eg, my ServiceFabric one), then this will not register  the default IMembershipOracle implementation as a SystemTarget. Doesn't matter yet.